crates/chain/src/keychain/txout_index.rs

```diff
@@ -206,9 +206,11 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     fn next_store_index(&self, keychain: &K) -> u32 {
         self.inner()
             .all_spks()
+            // This range is filtering out the spks with a keychain different than
+            // `keychain`. We don't use filter here as range is more optimized.
```

Comment:
I would prefer *"Range through all spks of `keychain`."*. However, I feel like this comment is not needed as the codebase already shows this.

Definitely not a blocker though.